src/components/Navigation/components/Item/Item.tsx

```diff
@@ -87,13 +88,27 @@ export function Item({
   const {isNavigationCollapsed} = useMediaQuery();
   const {location, onNavigationDismiss} = useContext(NavigationContext);
   const [expanded, setExpanded] = useState(false);
+  const [keyFocused, setKeyFocused] = useState(false);
 
   useEffect(() => {
     if (!isNavigationCollapsed && expanded) {
       setExpanded(false);
     }
   }, [expanded, isNavigationCollapsed]);
 
+  const handleKeyUp = useCallback(
+    (event) => {
+      if (event.keyCode === Key.Tab) {
+        !keyFocused && setKeyFocused(true);
+      }
+    },
+    [keyFocused],
+  );
+
+  const handleBlur = useCallback(() => {
+    keyFocused && setKeyFocused(false);
```

Comment:
Only set state if item was focused. This prevents calling set state on ever click off of a nav item